action-types.yml

```diff
@@ -0,0 +1,31 @@
+# See https://github.com/typesafegithub/github-actions-typing/
+inputs:
+  sudo:
+    type: boolean
```

Comment:
Well, this is not correct. The default is `auto`, which is not a Boolean value.

This actually points out a direction that I could get behind: using [input types in the `action.yml` file](https://github.blog/changelog/2021-11-10-github-actions-input-types-for-manual-workflows/). For example, the `sudo` input, which is currently declared [thusly](https://github.com/Vampire/action-tmate/blob/cdec350374bae61990f0a468a889a34afdb75d5f/action.yml#L12-L15):

```yaml
  sudo:
    description: 'If apt should be executed with sudo or without'
    required: false
    default: 'auto'
```

we could add the `type` and `options` attributes to define it like this instead:

```yaml
  sudo:
    description: 'If apt should be executed with sudo or without'
    required: false
    default: 'auto'
    type: choice
    options:
    - auto
    - false
    - true
```

This would likely achieve at least most of what `typesafegithub` seems to aim for?

Reply:
Oh, on second glance, it looks as if this is only for `workflow_dispatch` workflows, not for GitHub Actions' `action.yml` files (whose syntax [appears to be limited to `description`, `required`, `default` and `deprecationMessage`](https://docs.github.com/en/actions/sharing-automations/creating-actions/metadata-syntax-for-github-actions#inputs), something that [would appear unlikely to change](https://developers.slashdot.org/story/24/07/23/1915221/github-is-starting-to-feel-like-legacy-software)).

Reply:
Right, we could model it as an enum, github-actions-typing supports it.

Reply:
> Oh, on second glance, it looks as if this is only for workflow_dispatch workflows, not for GitHub Actions' action.yml files 

Exactly, it was requested in https://github.com/orgs/community/discussions/32054 that they add native typing information, but for the time being there is no definition or support for it.

The 3rd party typing information could of course also be stored within the action file, but it was considered a better idea to have it in a separate file in case GitHub makes up their mind and supports native typing which likely would then be in the action file so that there will not be a clash of the tow approaches.

> This would likely achieve at least most of what typesafegithub seems to aim for?

This could also be achieved with gat as @krzema12 said, using
```yaml
sudo:
  type: enum
  options:
    - auto
    - true
    - false
```
if you like.

I just interpreted the situation as a three-sate boolean.
Not having anything set behaves either like `true` or `false` depending on context.
That is what you call `auto`.
I thought if you set a value, you will set either `true` or `false` but never explicitly `auto`.
But I could of course change this to `enum` if you like.